subgraphs/collectible-market/mappings/index.ts

```diff
@@ -0,0 +1,263 @@
+import { Collection, NFT, Transaction, User } from "../generated/schema";
+import {
+  AskCancel,
+  AskNew,
+  CollectionClose,
+  CollectionNew,
+  CollectionUpdate,
+  RevenueClaim,
+  Trade,
+} from "../generated/CollectibleMarketV1/CollectibleMarketV1";
+
+import { toBigDecimal } from "./utils";
+
+import { IERC721 } from "../generated/CollectibleMarketV1/IERC721";
+
+let ZERO_BI = BigInt.fromI32(0);
+let ONE_BI = BigInt.fromI32(1);
+let ZERO_BD = BigDecimal.fromString("0");
+let EIGHTEEN_BD = BigDecimal.fromString("1e18");
+
+export function fetchCollectionName(collectionAddress: Address): string {
+  let contract = IERC721.bind(collectionAddress);
+  let nameValue = "unknown";
+  let nameResult = contract.try_name();
+
+  if (!nameResult.reverted) {
+    nameValue = nameResult.value;
+  }
+  return nameValue;
+}
+
+export function fetchCollectionSymbol(collectionAddress: Address): string {
+  let contract = IERC721.bind(collectionAddress);
+  let symbolValue = "unknown";
+  let symbolResult = contract.try_symbol();
+
+  if (!symbolResult.reverted) {
+    symbolValue = symbolResult.value;
+  }
+  return symbolValue;
+}
+
+export function fetchTokenURI(collectionAddress: Address, tokenId: BigInt): string {
+  let contract = IERC721.bind(collectionAddress);
+  let tokenURIValue = "unknown";
+  let tokenURIResult = contract.try_tokenURI(tokenId);
+
+  if (!tokenURIResult.reverted) {
+    tokenURIValue = tokenURIResult.value;
+  }
+  return tokenURIValue;
+}
+
+/**
+ * COLLECTION
+ */
+
+export function handleCollectionNew(event: CollectionNew): void {
+  let collection = Collection.load(event.params.collection.toHex());
+  if (collection == null) {
+    collection = new Collection(event.params.collection.toHex());
+    collection.name = fetchCollectionName(event.params.collection);
+    collection.symbol = fetchCollectionSymbol(event.params.collection);
+    collection.active = true;
+    collection.totalTrades = ZERO_BI;
+    collection.totalVolumeBNB = ZERO_BD;
+    collection.numberTokensListed = ZERO_BI;
+    collection.creatorAddress = event.params.creator.toHex();
+    collection.tradingFee = toBigDecimal(event.params.tradingFee, 2);
+    collection.creatorFee = toBigDecimal(event.params.creatorFee, 2);
+    collection.whitelistChecker = event.params.whitelistChecker.toHex();
+  } else {
+    // Collection has existed, was closed, but is re-listed
+    collection.active = true;
+    collection.creatorAddress = event.params.creator.toHex();
+    collection.tradingFee = toBigDecimal(event.params.tradingFee, 2);
+    collection.creatorFee = toBigDecimal(event.params.creatorFee, 2);
+    collection.whitelistChecker = event.params.whitelistChecker.toHex();
+  }
+  collection.save();
+}
+
+export function handleCollectionClose(event: CollectionClose): void {
+  let collection = Collection.load(event.params.collection.toHex());
+  collection.active = false;
+  collection.save();
```

Comment:
```suggestion
  let collection = Collection.load(event.params.collection.toHex());
  if (collection !== null) {
    collection.active = false;
    collection.save();
  }
```
Normally, this would be great and enough, unfortunately, given that we have this kind of [issue](https://status.thegraph.com/incidents/xlnq51ktp799) on TheGraph, I would 'strongly' suggest to ensure the entity isn't null before saving, otherwise the subgraph will crash and stop indexing blocks.